deployment/ccip/changeset/v1_6/cs_chain_contracts.go

```diff
@@ -71,209 +69,29 @@ var (
-					},
-				})
-			}
-			prevRampsTx, err = nm.ApplyPreviousRampsUpdates(txOpts, previousRampsArgs)
-			if cfg.MCMS == nil {
-				if _, err := cldf.ConfirmIfNoErrorWithABI(e.BlockChains.EVMChains()[chainSel], prevRampsTx, nonce_manager.NonceManagerABI, err); err != nil {
-					return cldf.ChangesetOutput{}, fmt.Errorf("error updating previous ramps for chain %s: %w", e.BlockChains.EVMChains()[chainSel].String(), err)
-				}
-			} else {
-				if err != nil {
-					return cldf.ChangesetOutput{}, fmt.Errorf("error updating previous ramps for chain %s: %w", e.BlockChains.EVMChains()[chainSel].String(), err)
-				}
-			}
-		}
-		if cfg.MCMS != nil {
-			mcmsTransactions := make([]mcmstypes.Transaction, 0)
-			if authTx != nil {
-				mcmsTx, err := proposalutils.TransactionForChain(chainSel, nm.Address().Hex(), authTx.Data(), big.NewInt(0),
-					string(shared.NonceManager), []string{})
-				if err != nil {
-					return cldf.ChangesetOutput{}, fmt.Errorf("failed to create transaction for chain %d: %w", chainSel, err)
-				}
-
-				mcmsTransactions = append(mcmsTransactions, mcmsTx)
-			}
-			if prevRampsTx != nil {
-				mcmsTx, err := proposalutils.TransactionForChain(chainSel, nm.Address().Hex(), prevRampsTx.Data(), big.NewInt(0),
-					string(shared.NonceManager), []string{})
-				if err != nil {
-					return cldf.ChangesetOutput{}, fmt.Errorf("failed to create transaction for chain %d: %w", chainSel, err)
-				}
 
-				mcmsTransactions = append(mcmsTransactions, mcmsTx)
-			}
-			if len(mcmsTransactions) == 0 {
-				return cldf.ChangesetOutput{}, errors.New("no operations to batch")
-			}
-
-			batches = append(batches, mcmstypes.BatchOperation{
-				ChainSelector: mcmstypes.ChainSelector(chainSel),
-				Transactions:  mcmsTransactions,
-			})
-
-			timelocks[chainSel] = s.Chains[chainSel].Timelock.Address().Hex()
-			inspectors[chainSel], err = proposalutils.McmsInspectorForChain(e, chainSel)
-			if err != nil {
-				return cldf.ChangesetOutput{}, fmt.Errorf("failed to get inspector for chain %d: %w", chainSel, err)
-			}
-		}
-	}
-	if cfg.MCMS == nil {
-		return cldf.ChangesetOutput{}, nil
-	}
-	mcmsContractByChain, err := deployergroup.BuildMcmAddressesPerChainByAction(e, s, cfg.MCMS)
+	seqReport, err := operations.ExecuteSequence(e.OperationsBundle, ccipseqs.UpdateNonceManagerSequence, deps, cfg)
 	if err != nil {
-		return cldf.ChangesetOutput{}, fmt.Errorf("error getting mcms contract by chain: %w", err)
-	}
-	proposal, err := proposalutils.BuildProposalFromBatchesV2(
-		e,
-		timelocks,
-		mcmsContractByChain,
-		inspectors,
-		batches,
-		"Update nonce manager for previous ramps and authorized callers",
-		*cfg.MCMS,
-	)
-	if err != nil {
-		return cldf.ChangesetOutput{}, err
+		return cldf.ChangesetOutput{}, fmt.Errorf("failed to execute UpdateNonceManager sequence: %w", err)
 	}
 
-	return cldf.ChangesetOutput{MCMSTimelockProposals: []mcmslib.TimelockProposal{*proposal}}, nil
+	return seqReport.Output.ToChangesetOutput(nil), nil
```

Comment:
Would use this instead: https://github.com/smartcontractkit/chainlink/blob/be2cfa4acb2194e9dc62fad69a13dc92f8b50238/deployment/ccip/shared/opsutil/evm.go#L99